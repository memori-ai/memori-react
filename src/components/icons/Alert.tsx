import React from 'react';

const Alert = ({
  className,
  title,
}: {
  className?: string;
  title?: string;
}) => (
  <svg
    width="800px"
    height="800px"
    viewBox="0 0 24 24"
    fill="none"
    {...(!title ? { 'aria-hidden': 'true' } : {})}
    xmlns="http://www.w3.org/2000/svg"
    className={className}
    aria-label={title}
    color="currentColor"
  >
    <path
      d="M12 16H12.01M12 8V12M12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21Z"
      stroke="currentColor"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
);

export default Alert;
