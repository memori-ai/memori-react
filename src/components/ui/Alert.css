.memori-alert {
    position: fixed;
    z-index: 50;
    top: 20px;
    right: 20px;
    width: var(--memori-alert--width, 400px);
  }
  
  .memori-alert--container {
    display: flex;
    align-items: flex-start;
    padding: 16px;
    border-radius: 8px;
    margin-bottom: 8px;
    background-color: white;
    box-shadow: 0 4px 6px -1px rgb(0 0 0 / 0.1), 0 2px 4px -2px rgb(0 0 0 / 0.1);
  }
  
  .memori-alert--icon {
    flex-shrink: 0;
    margin-right: 12px;
  }
  
  .memori-alert--content {
    min-width: 0;
    flex: 1;
  }
  
  .memori-alert--title {
    margin-bottom: 4px;
    font-size: 16px;
    font-weight: 600;
  }
  
  .memori-alert--description {
    color: rgb(107 114 128);
    font-size: 14px;
  }
  
  .memori-alert--actions {
    display: flex;
    align-items: center;
    margin-left: 16px;
  }
  
  .memori-alert--action {
    margin-right: 8px;
  }
  
  .memori-alert--close {
    flex-shrink: 0;
  }
  
  /* Alert types */
  .memori-alert--success {
    --alert-color: rgb(34 197 94);
    --alert-bg: rgb(240 253 244);
  }
  
  .memori-alert--warning {
    --alert-color: rgb(234 179 8);
    --alert-bg: rgb(254 252 232);
  }
  
  .memori-alert--error {
    --alert-color: rgb(239 68 68);
    --alert-bg: rgb(254 242 242);
  }
  
  .memori-alert--info {
    --alert-color: rgb(59 130 246);
    --alert-bg: rgb(239 246 255);
  }
  
  .memori-alert--success .memori-alert--container {
    border: 1px solid var(--alert-color);
    background-color: var(--alert-bg);
  }
  
  .memori-alert--warning .memori-alert--container {
    border: 1px solid var(--alert-color);
    background-color: var(--alert-bg);
  }
  
  .memori-alert--error .memori-alert--container {
    border: 1px solid var(--alert-color);
    background-color: var(--alert-bg);
  }
  
  .memori-alert--info .memori-alert--container {
    border: 1px solid var(--alert-color);
    background-color: var(--alert-bg);
  }