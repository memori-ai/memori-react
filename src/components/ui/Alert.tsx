import React, { FC } from 'react';
import { Transition } from '@headlessui/react';
import cx from 'classnames';
import Close from '../icons/Close';
import Button from './Button';
import { useTranslation } from 'react-i18next';

export type AlertType = 'success' | 'warning' | 'error' | 'info';

export interface Props {
  open?: boolean;
  onClose: (value: boolean) => void;
  className?: string;
  title?: string | JSX.Element | React.ReactNode;
  description?: string | JSX.Element | React.ReactNode;
  children?: JSX.Element | React.ReactNode;
  type?: AlertType;
  icon?: JSX.Element | React.ReactNode;
  closable?: boolean;
  duration?: number | null;
  action?: JSX.Element | React.ReactNode;
  width?: string;
}

const Alert: FC<Props> = ({
  open = false,
  onClose,
  className,
  title,
  description,
  children,
  type = 'info',
  icon,
  closable = true,
  duration = null,
  action,
  width = '400px',
}: Props) => {
  const { t } = useTranslation();

  // Auto-close functionality
  React.useEffect(() => {
    if (duration && open) {
      const timer = setTimeout(() => {
        onClose(false);
      }, duration);
      return () => clearTimeout(timer);
    }
  }, [duration, open, onClose]);

  const getTypeStyles = (type: AlertType) => {
    switch (type) {
      case 'success':
        return 'memori-alert--success';
      case 'warning':
        return 'memori-alert--warning';
      case 'error':
        return 'memori-alert--error';
      default:
        return 'memori-alert--info';
    }
  };

  return (
    <Transition
      show={open}
      as={React.Fragment}
      appear
    >
      <div className={cx('memori-alert', getTypeStyles(type), className)}>
        <style
          dangerouslySetInnerHTML={{
            __html: `
              .memori-alert {
                --memori-alert--width: ${width};
              }
            `,
          }}
        />
        <Transition.Child
          as={React.Fragment}
          enter="ease-out duration-300"
          enterFrom="opacity-0 translate-y-4"
          enterTo="opacity-100 translate-y-0"
          leave="ease-in duration-200"
          leaveFrom="opacity-100 translate-y-0"
          leaveTo="opacity-0 translate-y-4"
        >
          <div className="memori-alert--container">
            {icon && <div className="memori-alert--icon">{icon}</div>}
            
            <div className="memori-alert--content">
              {title && (
                <div className="memori-alert--title">{title}</div>
              )}
              {description && (
                <div className="memori-alert--description">{description}</div>
              )}
              {children}
            </div>

            <div className="memori-alert--actions">
              {action && <div className="memori-alert--action">{action}</div>}
              {closable && (
                <Button
                  ghost
                  padded
                  shape="circle"
                  icon={<Close />}
                  title={t('close') || 'Close'}
                  onClick={() => onClose(false)}
                  className="memori-alert--close"
                />
              )}
            </div>
          </div>
        </Transition.Child>
      </div>
    </Transition>
  );
};

export default Alert;