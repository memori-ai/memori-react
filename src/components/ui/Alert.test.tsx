import React from 'react';
import { render, screen, fireEvent, act } from '@testing-library/react';
import Alert from './Alert';
import Warning from '../icons/Warning';

beforeEach(() => {
  // Mock IntersectionObserver
  // @ts-ignore
  window.IntersectionObserver = jest.fn(() => ({
    observe: jest.fn(),
    unobserve: jest.fn(),
    disconnect: jest.fn(),
    takeRecords: jest.fn(),
  }));

  // Mock setTimeout/clearTimeout
  jest.useFakeTimers();
});

afterEach(() => {
  jest.useRealTimers();
});

describe('Alert Component', () => {
  it('renders Alert unchanged', () => {
    const { container } = render(
      <Alert
        open={false}
        onClose={jest.fn()}
        title="Test Alert"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders Alert open unchanged', () => {
    const { container } = render(
      <Alert
        open={true}
        onClose={jest.fn()}
        title="Test Alert"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('calls onClose when close button is clicked', () => {
    const onClose = jest.fn();
    render(
      <Alert
        open={true}
        onClose={onClose}
        title="Test Alert"
      />
    );
    
    const closeButton = screen.getByTitle('close');
    fireEvent.click(closeButton);
    expect(onClose).toHaveBeenCalled();
  });


  it('auto-dismisses after duration', () => {
    const onClose = jest.fn();
    render(
      <Alert
        open={true}
        onClose={onClose}
        title="Test Alert"
        duration={3000}
      />
    );

    act(() => {
      jest.advanceTimersByTime(3000);
    });

    expect(onClose).toHaveBeenCalled();
  });

  it('renders with custom width', () => {
    const { container } = render(
      <Alert
        open={true}
        onClose={jest.fn()}
        title="Test Alert"
        width="500px"
      />
    );
    
    const style = window.getComputedStyle(container.firstChild as Element);
    expect(style.getPropertyValue('--memori-alert--width')).toBe('500px');
  });

  it('renders with action button', () => {
    const actionClick = jest.fn();
    render(
      <Alert
        open={true}
        onClose={jest.fn()}
        title="Test Alert"
        action={<button onClick={actionClick}>Action</button>}
      />
    );
    
    const actionButton = screen.getByText('Action');
    fireEvent.click(actionButton);
    expect(actionClick).toHaveBeenCalled();
  });

  it('cleans up timer on unmount', () => {
    const onClose = jest.fn();
    const { unmount } = render(
      <Alert
        open={true}
        onClose={onClose}
        title="Test Alert"
        duration={3000}
      />
    );

    unmount();
    act(() => {
      jest.advanceTimersByTime(3000);
    });

    expect(onClose).not.toHaveBeenCalled();
  });
});