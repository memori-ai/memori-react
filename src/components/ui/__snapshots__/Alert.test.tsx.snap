// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Component renders Alert open unchanged 1`] = `
<div>
  <div
    class="memori-alert memori-alert--info"
  >
    <style>
      
              .memori-alert {
                --memori-alert--width: 400px;
              }
            
    </style>
    <div
      class="memori-alert--container ease-out duration-300 opacity-0 translate-y-4"
    >
      <div
        class="memori-alert--content"
      >
        <div
          class="memori-alert--title"
        >
          Test Alert
        </div>
      </div>
      <div
        class="memori-alert--actions"
      >
        <button
          class="memori-button memori-button--ghost memori-button--circle memori-button--padded memori-button--icon-only memori-alert--close"
          title="close"
        >
          <span
            class="memori-button--icon"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              role="img"
              viewBox="0 0 1024 1024"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                fill="currentColor"
                stroke="none"
                stroke-width="0"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert Component renders Alert unchanged 1`] = `<div />`;
